### ISSUE TITLE
Init apparently runs inside of kube-controller-manager

### ISSUE THREAD

**Issue by halkeye** [juliohm1978/kubernetes-cifs-volumedriver#13]
`./cifs init` apparently runs inside of kube-controller-manager which doesn't have /bin/bash, so 
```
kube-system/kube-controller-manager-nickfury[kube-controller-manager]: W0105 02:31:24.462231       1 driver-call.go:150] FlexVolume: driver call failed: executable: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/juliohm~cifs/cifs, args: [init], error: fork/exec /usr/libexec/kubernetes/kubelet-plugins/volume/exec/juliohm~cifs/cifs: no such file or directory, output: ""
```

switching to `#!/bin/sh` shebang gives us
```
kube-system/kube-controller-manager-nickfury[kube-controller-manager]: W0105 02:41:24.557182       1 driver-call.go:150] FlexVolume: driver call failed: executable: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/juliohm~cifs/cifs, args: [init], error: exit status 1, output: "-ne { \"status\": \"Failure\", \"message\": \"'jq' binary not found. Please install jq package before using this driver\"}\n"
```

Installing jq inside the container, moves onto the next issue

```
kube-system/kube-controller-manager-nickfury[kube-controller-manager]: E0105 02:47:02.950442       1 driver-call.go:267] Failed to unmarshal output for command: init, output: "-ne {\"status\": \"Success\", \"capabilities\": {\"attach\": false}}\n", error: invalid character 'n' in numeric literal
```

which is fixed by switching to /bin/echo instead of the built in echo

I can submit a PR that fixes some of these, but i'm not sure how to fix the jq inside of kube-controller-manager thing

**Comment by halkeye** [comment]
So I have a fork now with more detailed error trapping as its in go

https://github.com/halkeye/kubernetes-cifs-volumedriver/blob/master/main.go

I'm pretty sure you said in the past you were not interested in it, but lemme know if you want me to upstream it

**Comment by juliohm1978** [comment]
Nice! Thank you. I'll try to get working on this as soon as I can.

Meanwhile, there is a working v2.0-beta available from this branch:

https://github.com/juliohm1978/kubernetes-cifs-volumedriver/tree/golang

I finally got around to learning it. The driver was rewritten in Go, and it should avoid the problems from this issue. Feel free to give it a test run.

**Comment by juliohm1978** [comment]
I'll take a look at your branch as well. Thanks for the effort!

**Comment by juliohm1978** [comment]
Ok, I suppose this issue would be fixed by using either Go implementation.

Looking at your branch, a few differences come up:

* 3rd party dependencies, ie: `github.com/halkeye/flexvolume`.
* Missing patch for https://github.com/juliohm1978/kubernetes-cifs-volumedriver/issues/8, including support for k8s 1.14 & older versions (`kubernetes.io/fsGroup`).
* No need to write credentials to temporary file, that was only needed to keep bash from interpreting special characters.
* Unit tests.

Aside from those, it looks pretty solid. Keep in mind, I don't mean to critize. I'm happy to use either branch that works better in the long run.

**Comment by halkeye** [comment]
Yea, I like the version before I used that dep. I have a PR ready to add a few of the error handing i found to yours.

I'm way happier with yours

**Comment by juliohm1978** [comment]
Fixed by https://github.com/juliohm1978/kubernetes-cifs-volumedriver/pull/16, released with v2.0.
